kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -135,6 +187,85 @@ private MetricsRegistry metricsRegistry()   {
         }
     }
 
+    private boolean isBrokerState(MetricName name) {
+        return "BrokerState".equals(name.getName())
+                && "kafka.server".equals(name.getGroup())
+                && "KafkaServer".equals(name.getType());
+    }
+    private boolean isRemainingLogsToRecover(MetricName name) {
+        return "remainingLogsToRecover".equals(name.getName())
+                && "kafka.log".equals(name.getGroup())
+                && "LogManager".equals(name.getType());
+    }
+    private boolean isRemainingSegmentsToRecover(MetricName name) {
+        return "remainingSegmentsToRecover".equals(name.getName())
+                && "kafka.log".equals(name.getGroup())
+                && "LogManager".equals(name.getType());
+    }
+
+    private void startBrokerStateServer() throws Exception {
+        Server server = new Server();
+
+        HttpConfiguration https = new HttpConfiguration();
+        https.addCustomizer(new SecureRequestCustomizer());
+
+        ServerConnector conn = new ServerConnector(server,
+                new SslConnectionFactory(getSSLContextFactory(), "http/1.1"),
+                new HttpConnectionFactory(https));
+        conn.setPort(HTTPS_PORT);
+
+        ContextHandler context = new ContextHandler(BROKER_STATE_PATH);
+        Handler handler = new AbstractHandler() {
+            @Override
+            public void handle(String s, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException {
+                response.setContentType("application/json");
+                response.setCharacterEncoding("UTF-8");
+                baseRequest.setHandled(true);
+
+                Map<String, Object> brokerStateResponse = new HashMap<>();
+                if (brokerState != null) {
+                    if ((byte) brokerState.value() == 2 && remainingLogsToRecover != null && remainingSegmentsToRecover != null) {
```

Comment:
can't we have a constant instead of 2, as we have for `BROKER_RUNNING_STATE` and `BROKER_UNKNOWN_STATE`?